### DISCUSSION TITLE
Making Side-by-Side SBS usable on Steamdeck

### DISCUSSION THREAD

**Discussion by CarstenMaul** [wheaney/XRLinuxDriver#35] [Ideas]
Maybe this thread should go to the decky-xrealair repro but discussions are not enabled there.

I want to start a discussion about how to make SBS support usable without crashing your brain on Steamdeck and enjoy games in 3D SBS mode on Steamdeck/XREAL glasses.

"SBS" mode enables a 3840x1080 screen mode presented to the computer that the glasses split half-half to 1920x1080 streams streaming "half" of that ultrawide resolution to each of the two OLED projectors inside the XREAL glasses. This is being referred to as Side-by-Side full mode or non-streched (so-called in the game Trine)

Now there are two steps you have to master.

First, you must be able to configure the game to use SBS.
The Ultra-Wide 3840x1080 Screen Resolution must be recognized in up to 4 places, depending how the display config of the game is being designed:
- by the Steamdeck in the Displays Config
- by the Properties Setting in the SteamUI of the Game in the Screen Resolution Setting
- In the Game Startup Configuration Dialog
- in the in-game Display settings 

To have the SBS ultrawide resolution getting introduced and recognized by this chain of configs the first things to do is to 
- set the display resolution to "native" in the displays setting of the steam config, so the steamdeck will switch automatically to the maximum resolution of the display connected, in this case that of the XREAL glasses
- set the glasses to SBS mode by holding the brightness + button for 3 seconds and releasing the button after a chime sounds. Now the steamdeck will find a 3840x1080 resolution display....and things get wired

Now the Stemdeck recognizes the ultrawide display and the glasses are in SBS mode - splitting the Steam UI between both eyes. Unfortunately your brain is not SBS capable and you will have to close ether the right or left eyelids to respectively see half of the SteamUI in a way your brain can process the image :-)

Now you can start a SBS capable game like "Trine 1" and configure the resolution to 3840x1080 and choose the SBS non-stretched mode and start it.

The goal of this discussion should be to make using SBS games more easy via the XREAL driver and Decky ui plugin and to document the ways to enable SBS to games in Steam.

**Comment by wheaney** [comment]
The plugin itself can't do much more than it's already doing. Did you see the "tutorial" dialog that teaches you about SBS when you enable it? It discusses all these things.
